testcases/kernel/syscalls/write/write05.c

```diff
@@ -57,160 +57,127 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <sys/wait.h>
 #include <sys/mman.h>
-#include "test.h"
+#include "tst_test.h"
 
-void setup(void);
-void cleanup(void);
-
-char *TCID = "write05";
-int TST_TOTAL = 1;
 char filename[100];
 int fd;
 
 char *bad_addr = 0;
 
-int main(int argc, char **argv)
+static void verify_write(void)
 {
-	int lc;
-
 	char pbuf[BUFSIZ];
 	int pipefildes[2];
 	int status, pid;
 
-	tst_parse_opts(argc, argv, NULL, NULL);
-
-	/* global setup */
-	setup();
-
-	/* The following loop checks looping state if -i option given */
-	for (lc = 0; TEST_LOOPING(lc); lc++) {
-
-		/* reset tst_count in case we are looping */
-		tst_count = 0;
-
 //block1:
-		tst_resm(TINFO, "Enter Block 1: test with bad fd");
-		if (write(-1, pbuf, 1) != -1) {
-			tst_resm(TFAIL, "write of invalid fd passed");
-		} else {
-			if (errno != EBADF) {
-				tst_resm(TFAIL, "expected EBADF got %d", errno);
-			}
-			tst_resm(TPASS, "received EBADF as expected.");
+	tst_res(TINFO, "Enter Block 1: test with bad fd");
```

Comment:
Can we please get rid of the block nonsense?

The enter and exit block messages make the test too verbose for no added value, we print either PASS or FAIL per the testcase which is more than enough.